python/foglamp/plugins/south/cc2650/cc2650.py

```diff
@@ -0,0 +1,403 @@
+                'readings': {"x": gyro_x, "y": gyro_y, "z": gyro_z}
+            })
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, _handle['accelerometerSensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': {"x": acc_x, "y": acc_y, "z": acc_z}
+            })
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, _handle['magnetometerSensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': {"x": mag_x, "y": mag_y, "z": mag_z}
+            })
+
+        if _handle['batteryData']['value'] == 'true':
+            battery_level = tag.get_battery_level(
+                tag.char_read_hnd(_handle['characteristics']['battery']['data']['handle'], "battery"))
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, _handle['batterySensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': {"percentage": battery_level}
+            })
+    except RuntimeError as ex:
+        _plugin_restart(bluetooth_adr)
+        raise exceptions.QuietError(str(ex))
+    except (Exception, pexpect.exceptions.TIMEOUT) as ex:
+        _plugin_restart(bluetooth_adr)
+        raise exceptions.DataRetrievalError(str(ex))
+
+    return data
+
+
+def plugin_reconfigure(handle, new_config):
+    """  Reconfigures the plugin
+
+    it should be called when the configuration of the plugin is changed during the operation of the South service;
+    The new configuration category should be passed.
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+        new_config: JSON object representing the new configuration category for the category
+    Returns:
+        new_handle: new handle to be used in the future calls
+    Raises:
+    """
+    # In this method, cannot use "handle" as it might have changed due to restart. Hence use "_handle".
+    global _handle
+
+    bluetooth_adr = _handle['bluetoothAddress']['value']
+    _LOGGER.info("Old config for CC2650 {} plugin {} \n new config {}".format(bluetooth_adr, _handle, new_config))
+
+    # Find diff between old config and new config
+    diff = utils.get_diff(_handle, new_config)
+
+    # Plugin should re-initialize and restart if key configuration is changed
+    if 'bluetoothAddress' in diff or \
+                'assetPrefix' in diff or \
+                'maxRetry' in diff or \
+                'temperatureSensor' in diff or \
+                'luminanceSensor' in diff or \
+                'humiditySensor' in diff or \
+                'pressureSensor' in diff or \
+                'movementSensor' in diff or \
+                'batteryData' in diff:
+        plugin_shutdown(_handle)
+        new_handle = plugin_init(new_config)
+        new_handle['restart'] = 'yes'
```

Comment:
why do we need restart,  except bluetoothAddress, these should be polled and ingested with new config info